extensions/scarb-doc/src/docs_generation/markdown/traits.rs

```diff
@@ -14,12 +15,27 @@ pub trait TopLevelMarkdownDocItem: MarkdownDocItem + TopLevelDocItem {
         format!("{}.md", self.full_path().replace("::", "-"))
     }
 
-    fn md_ref(&self) -> String {
-        format!("[{}](./{})", self.name(), self.filename())
+    fn parent_path(&self) -> String {
+        let mut path_tree_elements: Vec<_> = self.full_path().split("::").collect();
+        path_tree_elements.pop();
+        path_tree_elements.join("::")
     }
 
-    fn generate_markdown_list_item(&self) -> String {
-        format!("- {}\n", self.md_ref())
+    fn md_ref(&self, display_parent_path: bool) -> String {
+        if display_parent_path {
+            format!(
+                "[{}  ({})](./{})",
+                self.name(),
+                self.parent_path(),
```

Comment:
Maybe instead of printing full path on duplicates, we could print only the path from the node where duplicate happens? Like in case of `hello:first::dup` and `hello::second::dup` we print `first::dup` and `second::dup`? WDYT?

Reply:
Hmmm, I think we could do that. Will work on that